README.md

```diff
@@ -1,7 +1,128 @@
 # KerasCV
-KerasCV is a repository of modular building blocks (layers, metrics, losses, data-augmentation) that applied computer vision engineers can leverage to quickly assemble production-grade, state-of-the-art training and inference pipelines for common use cases such as image classification, object detection, image segmentation, image data augmentation, etc. 
+KerasCV is a repository of modular building blocks (layers, metrics, losses, data-augmentation) that
+applied computer vision engineers can leverage to quickly assemble production-grade, state-of-the-art
+training and inference pipelines for common use cases such as image classification, object detection, 
+image segmentation, image data augmentation, etc. 
 
-KerasCV can be understood as a horizontal extension of the Keras API: they're new first-party Keras objects (layers, metrics, etc) that are too specialized to be added to core Keras, but that receive the same level of polish and backwards compatibility guarantees as the rest of the Keras API and that are maintained by the Keras team itself (unlike TFAddons).
+KerasCV can be understood as a horizontal extension of the Keras API: they're new first-party Keras 
+objects (layers, metrics, etc) that are too specialized to be added to core Keras, but that receive 
+the same level of polish and backwards compatibility guarantees as the rest of the Keras API and that 
+are maintained by the Keras team itself (unlike TFAddons).
 
-## Under Construction
-KerasCV is a relatively new project.  Expect regular updates and changes in the near future.
+# No Goals
+- **KerasCV is not a repository of blackbox end-to-end solutions (like TF Hub or ModelGarden).**
+
+    For the time being, it is focused on modular and reusable building blocks. We expect models in 
+    ModelGarden or TF Hub to eventually be built on top of the building blocks provided by KerasCV.
+
+    In the process of developing these building blocks, we will by necessity implement end-to-end 
+    workflows, but they're intended purely for demonstration and grounding purposes, they're not 
+    our main deliverable.
+
```

Comment:
In this block i think it is not clear where where you will store the e2d network, its training scripts or they pre-trained weights?
I think it is really important to understand by a community perspective where the trainable model and its weights will be stored.

Cause, in the current form, it seems that model gardens/TFHUB at some point will need to duplicate again all the e2e networks already implemented somewhere to test Keras-cv model driven components.

So I suppose that this is going to clarify where to contribute reusable components but not the e2e network.

Reply:
> In this block i think it is not clear where where you will store the e2d network, its training scripts or they pre-trained weights?

In short, both.  we intend to ensure reproducibility of the weights published alongside models in the package.   We will have `keras_cv.applications.*`, which will host applications similar to `keras_cv.applications`.   Certain applications (i.e. ResNet50) could also considered critical building blocks for many model architectures people build. 

> So I suppose that this is going to clarify where to contribute reusable components but not the e2e network.

We are happy to host a widely reusable network architecture (i.e. ResNet50 if it were not in core Keras) in keras_cv.  I don't think that having a duplicated architecture between us and model garden will dissuade us from hosting a reusable architecture.

Reply:
Ok but I still don't understand why we need to compete with models garden on the new model. Can we just figure out a shared way to add this models to Tensorflow garden?

I don't think that having duplicated training scripts, duplicated weights storages and hash  and duplicated models requests handling its is going to optimize what we always claim:
limited internal and community resoruces.

Reply:
> Ok but I still don't understand why we need to compete with models garden on the new models.  Can we just figure out a shared way to add this models to Tensorflow garden?

I think we want to provide something unique here.  In particular, we want to really focus on providing a coherent Keras ecosystem.

Reply:
I am totally ok but also with this claim I see a little bit of competitive forking between what we call now "Keras ecosystem" and something that was called as whole "TF ecosystem".
If the goal is to create this new entity "Keras ecosystem" I will not push more comments for a more holistic approach to coordinate with what we already have in the TF ecosystem and SIGs.

Reply:
> Ok but I still don't understand why we need to compete with models garden on the new models. Can we just figure out a shared way to add this models to Tensorflow garden?
> 
> I don't think that having duplicated training scripts, duplicated weights storages/hashes and duplicated models requests handling its is going to optimize what we always claim: limited internal and community resoruces.

Model garden has their opinionated way for implementing models/layers/training logic (Orbit), which is not necessary same as the way keras team would like to take. Forcing them to adopt the keras team process might be a big ask for them. We would like to provide our solution with keras best practice, and let model garden adopt it when we are ready. In this process, these components can be used by end user as well.

Reply:
I want to reinforce the claim that I am one of the promoter or reusable components. But if we need to create a competitive solution I will not going to push for more coordination.

Reply:
I think that we could resolve this as now we have a de-facto competitive call for contribution in the communiy and we could not reduce this spending in extra coordination.

Reply:
I still think that we could still expand this section with the clarification we had in this thread.

Reply:
I will add @martin-gorner and @fchollet in this thread to comment on the interaction between keras-team and model garden as well.

Reply:
Thanks, I am ok also if the final outcome will be to compete over concurrent philosophies, at that point we could just describe here our different point of view and then users will select what they want to use and the contributors will choice on which repo the will want to spent their time.

Reply:
Just to have a quick pointer to a recent overlap between models grader contribution proposal and keras-cv. I have tried to manually route the contributor with a comment:
https://github.com/keras-team/keras/issues/15545#issuecomment-1025728605

If we are in a competitive setup on model collection between keras-* and models-garden I don't think that we could route users with comment like the one I've done. 
If we disambiguate a little bit what we want to do with the new models it will be easier to a more clean feedback to users on github and on the forum.